### DISCUSSION TITLE
Domain and subdomain data aggregation

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#160] [Anything Else]
"A nice feature would be if you can set a domain and subdomains for tracking and see a link between them and/or aggregating of data tab?"

**Update:** You can now filter your traffic by hostname! This way you can get a global overview of traffic across all your hostnames as well as segment by a single subdomain all in the same Plausible dashboard. See more: https://plausible.io/docs/subdomain-hostname-filter

**Comment by mehulkar** [comment]
From #205, my use case is the _same_ site deployed on multiple domains, and wanting to see which one is used more.

**Comment by vermorel** [comment]
I operate multiple websites on the same domain: `www.lokad.com`, `blog.lokad.com`, `docs.lokad.com`. I would really like to have some control on the originating subdomain, as least to filter.

**Comment by shaunchurch** [comment]
I run one app that serves different content on many domains and would really like to be able to get a global overview of traffic across all the domains, as well as filter by a single domain.

**Comment by divinerites** [comment]
> I operate multiple websites on the same domain: `www.lokad.com`, `blog.lokad.com`, `docs.lokad.com`. I would really like to have some control on the originating subdomain, as least to filter.

In fact when you give the *domain name* in Plausible, it is just an ID for tracking.

So in your use case (tracking separate subdomains), i would just create 3 different ids, and treat/use each of them in their respective pages from their (sub)domain.
But you'll loose the aggregation.

**Comment by ltalirz** [comment]
Would be very useful for running PaaS instances like [dokku](http://dokku.viewdocs.io/dokku/) where you have one subdomain per app, and you would like for the number of subdomains to change dynamically

**Comment by shafy** [comment]
I have different subdomains for different languages of my website (e.g. yeticheese.com and de.yeticheese.com) - would be cool to see aggregate data. A related issue I'm having is that my auto language switcher redirects based on browser language, so people who go on yeticheese.com with a German browser are redireted to de.yeticheese.com and I then use the origin referrer. I know that I can do a workaround with the ref URL param, but still :-)

**Comment by codeofsumit** [comment]
Same here. We track our Web App (app.domain.com), Marketing App (domain.com) and Community (community.domain.com) and would love to filter them.

**Comment by bmenant** [comment]
:+1: We too would love to tack multi-domains (or at least any subdomains) on a same site set up on Plausible.

We’ve got a platform setup with dozens of subdomains (where each subdomain relates to a _campaign_ or an _operation_). As a matter of fact, each campaign looks pretty much like a dedicated website and we certainly do want to break down audience data per domain (i.e. per operation). The only thing is that we’ve got more than a hundred of those, as we can set up those very easily on the platform, like you’d publish a post on a blog... so we don’t want to and we cannot really create a new website setting for each of those.

Up until now, we use Plausible on our main website, but not on the platform, for this very reason.

- To have a global overview of what happens on the platform as a whole is desirable.
- To have a filtered view of what happens for each operation (i.e. domain) is as much as desirable. 

I’ve made a suggestion in #478

**Comment by kjgarza** [comment]
I would much prefer if an option such as having `<script data-site-id="<numeric-id">` rather than the `data-domain` attribute. It would make it way much straightforward for the user of the tracker.

**Reply by ukutaht** [reply]
Yes that's a change we're planning to do

**Reply by mannp** [reply]
> Yes that's a change we're planning to do

I am about to redo my sites that plausible monitors as I've originally entered the base domain and am not able to track the subdomains separately, so I wondered if the planned change would require a reset of the config anyway or whether we'd be able to keep the data we have already?

Thanks.

**Reply by metmarkosaric** [reply]
the work on this hasn't started yet so it's difficult to say much but we will for sure do whatever is possible not to break anything so i wouldn't worry about that

**Reply by mannp** [reply]
> the work on this hasn't started yet so it's difficult to say much but we will for sure do whatever is possible not to break anything so i wouldn't worry about that

I see, thanks for the clarification.

**Comment by RobertJoonas** [comment]
Feedback via email:

> I’m tracking a lot of websites and subdomains with one tracking code. I want to get an overview of all websites published by us. The Problem is, I can’t see the Domain in the Dashboard, so I can’t filter by domain.
> Is it possible to show that in the results. Something like tracking with an ID?

**Comment by metmarkosaric** [comment]
Leaving a note here for those that find this thread when looking for possible solutions to the use case of wanting multiple site stats aggregated in one dashboard:

We do support roll-up reporting. It allows you to aggregate stats from multiple sites and see them all together in one dashboard while still keeping the individual site stats on their own separate dashboards. Here's [how you can use it](https://plausible.io/docs/plausible-script#is-there-a-roll-up-view).

We do also plan to support the "all in one" dashboard only use case in the future but the roll-up may be useful to some in the meanwhile.

**Reply by svogt0511** [reply]
Has this has been merged to master in the plausible repository? I have a forked repository and would like to give it a try.  Thanks.

**Reply by metmarkosaric** [reply]
yep, should be a part of the latest release

**Reply by humanagent** [reply]
The roll-up works well but im not able to see which domain the page corresponds to. 
So if i have mutiple domains the "/" source is general so top pages data is useless for now.

**Comment by fostermh** [comment]
Any idea when the "all in one" dashboard feature will be implemented? It seems to me that https://github.com/plausible/analytics/pull/478 fixed the domain issue quite nicely with the exception of a minor display issue. I would quite like to move a multi organization / multi site project away from GA to using Plausible. However, the lack of ability to filter by domain or even see which domain a page request originates from is a major blocker.

**Comment by arifb** [comment]
Hi - am I correct to say that tracking an app with multiple subdomains is not possible with plausible? Or, said differently, you can't track traffic to each of the subdomains as the tool will only show you the relative path of the page visited, rather than by domain?

Is not possible to send a custom event or similar to append the subdomain to the path, similar to how goat counter does it? For example:
```
    window.goatcounter = {
        path: function(p) { return location.host + p }
    }
```

**Reply by metmarkosaric** [reply]
hi @arifb! the best way to track multiple subdomains on the same dashboard but also see the page paths by domain at this stage is to use our roll-up reporting. see https://plausible.io/docs/plausible-script#is-there-a-roll-up-view

we do allow you to send custom dimensions alongside custom events so you could try and explore if that does what you need too https://plausible.io/docs/custom-event-goals#using-custom-props

**Reply by arifb** [reply]
Thanks for the quick reply @metmarkosaric. If we were to implement the rollup view, would it be easy to parse out which subdomain is contributing to, for example, the top pages list? Or would it just be a relative path, similar to https://plausible.io/plausible.io/pages?

Our use case is we have many subdomains, all with the same relative routes ie. `subdomain1.site.com/stories` and `subdomain2.site.com/stories`. Would I just see `/stories` in the rollup view?

Basically we want to be able to use one view / plausible UI to analyze traffic across all the properties, but ideally would like to see which subdomain is contributing the most to the stats. Otherwise, we'd have to visit each of the dashboards to parse it out.

Thanks!

**Reply by metmarkosaric** [reply]
on the roll-up dashboard you can see all the combined traffic with relative paths only while you'd need to go to the individual site dashboard for the stats of that particular site

**Reply by arifb** [reply]
Thanks @metmarkosaric. 

Consider this a vote for what I thought would be a common use case: A tenanted by subdomain app that has the same relative paths across subdomains that is able to have 1 plausible dashboard for the whole app that considers each subdomain separately. ie. `subdomain1.site.com/stories` analytics is separate from `subdomain2.site.com/stories` but you're able to view both in 1 plausible dashboard.

Last thing - have you seen folks use custom props to achieve this?

**Reply by metmarkosaric** [reply]
thanks for the feedback! in most cases people are happy with having these on their own dashboards and then roll-up for those that need combined view too but yeah we've got this on our list of features we want to build in the future. custom props can be used to send custom data with custom events so yeah you could send a specific site name as a prop with each event conversion. we're also putting the finishing touches to custom props for pageviews which then will allow you to send custom data with any pageview. stay tuned!

**Comment by clementmas** [comment]
Are you planning to add a "Multi-domain" option like Fathom: https://usefathom.com/docs/features/multi?

I'm moving all my sites away from GA but some of them use several domain/subdomains so if it's not supported I won't be able to use Plausible.

I looked at the "Roll-up view" option but duplicating all the pageviews and having several dashboard is not an interesting option. Adding a "Domain" tab would be the way to go.

**Comment by metmarkosaric** [comment]
**Update:** You can now filter your traffic by hostname! This way you can get a global overview of traffic across all your hostnames as well as segment by a single subdomain all in the same Plausible dashboard. See more: https://plausible.io/docs/subdomain-hostname-filter
